src/commands.cpp

```diff
@@ -1457,67 +1464,75 @@ bool CommandMoveNode::procresult(Result r)
-                                                            n->nodehandle, 1, n->owner, ACCESS_UNKNOWN,
-                                                            0, NULL, NULL, it->first, false));
-                        }
-                    }
-
-                    if (n->outshares)
-                    {
-                        for (it = n->outshares->begin(); it != n->outshares->end(); it++)
-                        {
-                            client->newshares.push_back(new NewShare(
-                                                            n->nodehandle, 1, it->first, ACCESS_UNKNOWN,
-                                                            0, NULL, NULL, UNDEF, false));
-                        }
-                    }
-
-                    client->mergenewshares(1);
-                }
-            }
+            // todo: is this really done by the actinpacket code
+
+            //Node *n = client->nodebyhandle(h);
+            //if (n && (n->pendingshares || n->outshares))
+            //{
+            //    Node *rootnode = client->nodebyhandle(np);
+            //    while (rootnode)
+            //    {
+            //        if (!rootnode->parent)
+            //        {
+            //            break;
+            //        }
+            //        rootnode = rootnode->parent;
+            //    }
+            //    if (rootnode && rootnode->type == RUBBISHNODE)
+            //    {
+            //        share_map::iterator it;
+            //        if (n->pendingshares)
+            //        {
+            //            for (it = n->pendingshares->begin(); it != n->pendingshares->end(); it++)
+            //            {
+            //                client->newshares.push_back(new NewShare(
+            //                                                n->nodehandle, 1, n->owner, ACCESS_UNKNOWN,
+            //                                                0, NULL, NULL, it->first, false));
+            //            }
+            //        }
+
+            //        if (n->outshares)
+            //        {
+            //            for (it = n->outshares->begin(); it != n->outshares->end(); it++)
+            //            {
+            //                client->newshares.push_back(new NewShare(
+            //                                                n->nodehandle, 1, it->first, ACCESS_UNKNOWN,
+            //                                                0, NULL, NULL, UNDEF, false));
+            //            }
+            //        }
+
+            //        client->mergenewshares(1);
+            //    }
+            //}
         }
         else if (syncdel == SYNCDEL_NONE)
         {
             client->sendevent(99439, "Unexpected move error", 0);
         }
     }
-    client->app->rename_result(h, r.errorOrOK());
-    return r.wasErrorOrOK();
+
+    if (Node* n = client->nodebyhandle(h))
+    {
+        client->rewriteforeignkeys(n);
```

Comment:
This is a tricky one.  I had left the call there because I wasn't sure of the consequences of removing it at that time.  Looking at it now, the actionpacket code only calls rewriteforeignkeys() when processing mergenewshares().   I don't think it's ideal to have every logged in client (for this user) sending the key rewrites, it's better to just have the one client that initiated the move do that.  Though, it would be even better if that happened in a single all-or-nothing (strong guarantee) batch along with the move itself.   But I think that is out of the scope for this change, I think best to leave this here for this PR (as that's no worse than current behaviour)?  thanks

Reply:
Hmmm, interesting. I agree the best is to have a single client to rewrite the keys (the one that initiated the move). And that the better approach would be to do it as part of the command. Perhaps that's something to reconsider as part of the task to restore the key's healing mechanism (which is currently broken).
Anyway, I agree with you, we can leave this here for this PR ;)